### PR TITLE
Jwt authentication

### PR DESCRIPTION
**Pull request by kaloyan-st**

### CODE REVIEW

File: AI_CV_JAVA/pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/pom.xml:37]: Fix empty lines for the whole pom.xml file

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/DTO/UserDTO.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserDTO {
+
+    private int id;
+
+    private String firstname;
+
+    private String lastname;
+
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/DTO/UserDTO.java:17]: Add validation for the email and the other fields. Read about @Email, @Size, @NotNull and whether or not it is recommended to use validations both in the dto and the entity

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationRequest.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class AuthenticationRequest {
+
+    private String email;
+    String password;
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationRequest.java:16]: missed access modifier

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationResponse.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationResponse.java:14]: unnecessary empty line

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationService.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+
+    private final UserRepository repository;
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationService.java:17]: change variable name to userRepository

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationController.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/auth")
+@RequiredArgsConstructor
+public class AuthenticationController {
+
+    private final AuthenticationService service;
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationController.java:17]: change variable name to authenticationService

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationService.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        request.getEmail(),
+                        request.getPassword()
+                )
+        );
+        var user = repository.findByEmail(request.getEmail())
+                .orElseThrow();
+        var jwtToken = jwtService.generateToken(user);
+        return AuthenticationResponse.builder()
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/auth/AuthenticationService.java:47]: it is advisory to have an empty line before a return statement

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/config/UserConfig.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+@Configuration
+public class UserConfig {
+
+    @Bean
+    public UserService userBean() {
+        return new UserService();
+    }
+
+    @Bean
+    public ModelMapper modelMapperBean() {
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/config/UserConfig.java:18]: modelMapperBean -> modelMapper, userBean -> userService

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/controller/GoogleLoginController.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        this.googleLoginService = googleLoginService;
+    }
+
+    @PostMapping("/process-google-token")
+    public ResponseEntity<String> processGoogleToken(@RequestBody String googleToken) {
+        return googleLoginService.processGoogleToken(googleToken);
+    }
+}
+
+
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/controller/GoogleLoginController.java:24]: empty lines

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/ApplicationConfig.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfig {
+
+    private final UserRepository repository;
+    @Bean
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/ApplicationConfig.java:27]: empty line before the @Bean in order to separate it from the fields

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/JwtAuthenticationFilter.java [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.io.IOException;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtService jwtService;
+    private final UserDetailsService userDetailsService;
+    @Override
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/JwtAuthenticationFilter.java:25]: same here - separate the fields from the method with an empty line

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/JwtService.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class JwtService {
+    @Value("${jwt.secret-key}")
+    private String SECRET_KEY;
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/JwtService.java:22]: leave an empty line between fields which are annotated

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/JwtService.java [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver){
+        final Claims claims = extractAllClaims(token);
+        return claimsResolver.apply(claims);
+    }
+
+    public String generateToken(UserDetails userDetails) {
+        return generateToken(new HashMap<>(), userDetails);
+    }
+
+    // Generate a token
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/JwtService.java:40]: Remove comments

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/GoogleLoginService.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+    private final UserRepository userRepository;
+    private final JwtService jwtService;
+
+    public ResponseEntity<String> processGoogleToken(String googleToken) {
+        try {
+            GoogleIdToken idToken = verifyGoogleToken(googleToken);
+            Payload payload = idToken.getPayload();
+
+            String userId = payload.getSubject();
+            System.out.println("User ID: " + userId);
```
Comment (PetkoLyuts) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/GoogleLoginService.java:34]: System.out.prints should be removed

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/SecurityConfiguration.java [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class SecurityConfiguration {
+
+    private static final String[] WHITE_LIST_URL = {"/api/v1/auth/**",
+            "/process-google-token",
+            "/users/{id}"
```
Comment (mkanevv) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/security/SecurityConfiguration.java:22]: shouldn't be whitelisted

File: AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/GoogleLoginService.java [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class GoogleLoginService {
+
+    private final UserRepository userRepository;
+    private final JwtService jwtService;
+
+    public ResponseEntity<String> processGoogleToken(String googleToken) {
```
Comment (mkanevv) [inline AI_CV_JAVA/src/main/java/com/example/AI_CV_JAVA/service/GoogleLoginService.java:28]: No service should return a response entity

**Review by PetkoLyuts — APPROVED** [review]

**Pull request merged**